controller/scheduler/job.go

```diff
@@ -87,6 +87,10 @@ type Job struct {
 	// referenced from within the main scheduler loop
 	state JobState
 
+	// reason is a human readable indication of why a job is currently
+	// pending. It is only valid while the job is in the pending state.
+	reason string
```

Comment:
I think this should be something like `stateReason` and contain the reason the job is in any of the possible states, not just `pending` (e.g. a job is `down` because it either failed to start or started then exited).